src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -846,20 +853,43 @@ public SCM build(SCMHead head, SCMRevision revision) {
                 GitSCM scm = (GitSCM) super.build(head, revision);
                 String repoUrl = repositoryUrl(getRepoOwner(), getRepository());
                 if (repoUrl != null) {
-                    scm.setBrowser(new GithubWeb(repoUrl));
+                    setBrowser(scm, repoUrl);
                 }
                 return scm;
             }
         } else {
             GitSCM scm = (GitSCM) super.build(head, /* casting just as an assertion */(SCMRevisionImpl) revision);
             String repoUrl = repositoryUrl(getRepoOwner(), getRepository());
             if (repoUrl != null) {
-                scm.setBrowser(new GithubWeb(repoUrl));
+                setBrowser(scm, repoUrl);
             }
             return scm;
         }
     }
 
+    // TODO remove and replace with scm.setBrowser(repoUrl) directly once baseline Git plugin 3.0.2+
+    private void setBrowser(GitSCM scm, String repoUrl) {
+        try {
+            scm.setBrowser(new GithubWeb(repoUrl));
+        } catch (NoSuchMethodError e) {
+            Level level;
+            long now = System.currentTimeMillis();
+            long next = jenkins41244Warning.get();
+            if (now >= next) {
+                long newNext = now + TimeUnit.MINUTES.toMillis(5);
+                if (jenkins41244Warning.compareAndSet(next, newNext)) {
```

Comment:
Better to leave it to log handlers to decide how to deal with duplicated warnings.